x-pack/plugins/siem/server/lib/timeline/routes/update_timelines_route.ts

```diff
@@ -44,37 +43,47 @@ export const updateTimelinesRoute = (
         const frameworkRequest = await buildFrameworkRequest(context, security, request);
         const { timelineId, timeline, version } = request.body;
         const { templateTimelineId, templateTimelineVersion, timelineType } = timeline;
-        const isHandlingTemplateTimeline = timelineType === TimelineType.template;
-        const existTimeline =
-          timelineId != null ? await getTimeline(frameworkRequest, timelineId) : null;
 
-        const existTemplateTimeline =
-          templateTimelineId != null
-            ? await getTemplateTimeline(frameworkRequest, templateTimelineId)
-            : null;
-        const errorObj = checkIsFailureCases(
-          isHandlingTemplateTimeline,
-          version,
-          templateTimelineVersion ?? null,
-          existTimeline,
-          existTemplateTimeline
-        );
-        if (errorObj != null) {
-          return siemResponse.error(errorObj);
-        }
-        const updatedTimeline = await createTimelines(
-          (frameworkRequest as unknown) as FrameworkRequest,
-          timeline,
-          timelineId,
-          version
-        );
-        return response.ok({
-          body: {
-            data: {
-              persistTimeline: updatedTimeline,
-            },
+        const timelineStatus = new TimelinesStatus({
+          timelineType: timelineType ?? TimelineType.default,
```

Comment:
`TimelineStatus` has already `TimelineType.default` as a default value for `timelineType`, so propably it's redundant

Reply:
Since the timelineType here is from user input, and we are using `SavedTimelineRuntimeType`
x-pack/plugins/siem/common/types/timeline/index.ts here, which makes timelineType likely to be null, so I can't think of other ways to make type script happy but to put a fallback here.